### PR TITLE
fix(managers/nix): pass github token to nix

### PR DESCRIPTION
**Pull request by JamieMagee**
## Changes

Pass GitHub token to nix for lockfile updates

## Context

- Fixes #21567

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/modules/manager/nix/artifacts.ts [hunk 1]
```diff
@@ -20,23 +21,28 @@ export async function updateArtifacts({
...
+    --extra-experimental-features nix-command \
+    --extra-experimental-features flakes `;
+
+  const { token } = hostRules.find({
+    hostType: 'github',
+    url: 'https://api.github.com/',
+  });
+
+  if (token) {
+    cmd += `--extra-access-tokens github.com=${token} `;
```
Comment (viceice) [inline lib/modules/manager/nix/artifacts.ts:34]: i would prefer to pass it via `NIX_CONFIG` env 🤔

Comment (JamieMagee) [reply lib/modules/manager/nix/artifacts.ts:34]: That's okay for the GitHub app, but harder to do if someone is self-hosting. This way is potentially worse for logging, if you log shell commands, right?

Comment (rarkins) [reply lib/modules/manager/nix/artifacts.ts:34]: Why is env harder for self-hosting?

Comment (JamieMagee) [reply lib/modules/manager/nix/artifacts.ts:34]: Not env in general, but if someone has configured `NIX_CONFIG` we might overwrite it, or make it invalid.

**Review by viceice — APPROVED** [review]

**Pull request merged**